egs/mobvoi/v1/local/chain/run_tdnn_1a.sh

```diff
@@ -0,0 +1,314 @@
+#!/bin/bash
+
+
+# local/chain/compare_wer.sh exp/chain/e2e_tdnn_1a
+
+# steps/info/chain_dir_info.pl exp/chain/e2e_tdnn_1a
+# exp/chain/e2e_tdnn_1a: num-iters=102 nj=2..5 num-params=3.7M dim=40->84 combine=-0.117->-0.116 (over 3) logprob:train/valid[67,101,final]=(-0.080,-0.073,-0.073/-0.090,-0.089,-0.088)
+
+set -e
+
+# configs for 'chain'
+stage=0
+nj=30
+gmm=mono
+train_stage=-5 # starting from -5 to skip phone-lm estimation
+get_egs_stage=-10
+affix=1a
+remove_egs=false
+xent_regularize=0.1
+
+# training options
+srand=0
+num_epochs=3
+num_jobs_initial=2
+num_jobs_final=5
+chunk_width=140,100,160
+common_egs_dir=
+reporting_email=
+dim=450
+frames_per_iter=3000000
+bs_scale=0.0
+train_set=train
+test_sets="dev eval train"
+wake_word="嗨小问"
+
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging
+
+. ./cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+local/nnet3/run_ivector_common.sh \
+  --stage $stage --nj $nj \
+  --train-set $train_set --gmm $gmm || exit 1;
+
+gmm_dir=exp/${gmm}
+ali_dir=exp/${gmm}_ali_${train_set}_sp
+lat_dir=exp/chain/${gmm}_${train_set}_sp_lats
+train_data_dir=data/${train_set}_sp_hires
+train_ivector_dir=exp/nnet3/ivectors_${train_set}_sp_hires
+lores_train_data_dir=data/${train_set}_sp
+
+lang=data/lang_chain
+lang_decode=data/lang_chain_decode
+tree_dir=exp/chain/tree  # it's actually just a trivial tree (no tree building)
+dir=exp/chain/tdnn_${affix}
+
+for f in $train_data_dir/feats.scp $train_ivector_dir/ivector_online.scp \
+    $lores_train_data_dir/feats.scp $gmm_dir/final.mdl \
+    $ali_dir/ali.1.gz; do
+  [ ! -f $f ] && echo "$0: expected file $f to exist" && exit 1
+done
+
+if [ $stage -le 7 ]; then
+  echo "$0: creating lang directory $lang with chain-type topology"
+  # Create a version of the lang/ directory that has one state per phone in the
+  # topo file. [note, it really has two states.. the first one is only repeated
+  # once, the second one has zero or more repeats.
+  if [ -d $lang ]; then
+    if [ $lang/L.fst -nt data/lang/L.fst ]; then
+      echo "$0: $lang already exists, not overwriting it; continuing"
+    else
+      echo "$0: $lang already exists and seems to be older than data/lang ..."
+      echo " ... not sure what to do.  Exiting."
+      exit 1;
+    fi
+  else
+    rm -rf $lang
```

Comment:
`rm -rf $lang` can be deleted.
the `else` part is executed only when `[ -d $lang ]` fails, which means `$lang` does
not exist; so there is no need to delete it.